pairing/adapter.go

```diff
@@ -0,0 +1,47 @@
+package pairing
+
+import (
+	"github.com/dedis/kyber"
+	"github.com/dedis/kyber/pairing/bn256"
+)
+
+// SuiteBn256 is an adapter from the different bn256 suites to a single one
+// that can be used by onet to encode/decode key pairs. The public key comes
+// from the field G2 and the private key comes from G1 thus the Point and
+// Scalar functions return the corresponding type.
+type SuiteBn256 struct {
+	Suite
+	kyber.Group
```

Comment:
why do you need the `kyber.Group`?

Reply:
To implement the suites.Suite interface that has the .Point() function used to create public key points. Suite comes from the pairing suite that supports three kinds of suite but the interface assumes only one. That makes the suite compatible with the code that is using the interface.

Reply:
shut me down cause I can't resist: that is a hack producing wrong semantics (there's no such things as a bn256 point) because of a problem NOT in kyber ... :(
Why kyber should suffer for the wrongness in other repos while we spent time and effort making it good ... :/
One could imagine using these interface's ID @Gilthoniel is doing in protobuf for `bn256_G1` `bn256_G2` and `bn256_GT`, wouldn't it solve the problem ... ?

Reply:
this is a good point.. maybe we need to re-think what the pairing suite should look like.

I think the reason we need this kind of "standard" suite is because when onet/cothority registers one, it's one suite per service. What are the consequences if we register `pairing.Suite` @Gilthoniel ? It should be possible to test the suite implementation using the casting trick in go.

Reply:
Actually as @kc1212 pointed out, scalars are the same in both groups so we could use G2 as the public suite because G1 is used only internally when signing and verifying... I'm gonna try that tomorrow and see if it works.

Reply:
Ok now I remember why I did an adapter. You still need functions from the parent suite (e.i. the Pair function) then the purpose of the adapter was to keep the properties of the parent suite but make it compatible with the common suite interface.

I understand your worries about the point concept but in order to make it work, you need to have a common base to generate public and private keys. Take the key pair generator that takes a suites.Suite and thus expect Point and Scalar.

I'm not saying you're wrong and that's the only way. I guess one solution could be to rethink the suite interface so that it implements basic tools required like a key pair generation, sign, verify etc ... and hide any point logic from outside. But my point here is to make something that won't deeply change kyber.

Reply:
> Ok now I remember why I did an adapter. You still need functions from the parent suite (e.i. the Pair function) then the purpose of the adapter was to keep the properties of the parent suite but make it compatible with the common suite interface.

It's too late now but if one is using such functionalities `Pairing` then it knows it is using a pairing suite, therefore could it not retrieve the corresponding `pairing.Suite` from a given point or scalar ? i.e. go `bn256_G1` to `bn256.Suite`. 

Anyway... :(